File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int l = 0, r = nums.size() - 1;
        
        while (l < r) {
            int m = l + (r-1) / 2 //to prevent overflow for big l and r
            if (nums[mid] > nums[r]) {
                l = m + 1;
            } else {
                r = m;
            }
        }
        return num[l];
    }
};

File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)

        def jump(i):
            if i == n-1:
                return True
            
            if i >= n:
                return False
            
            jump_amount = nums[i]
            if jump_amount == 0 and i != n:
                return False
            
            return jump(i + jump_amount)
        
        return jump(0)